pkg/storage/cache.go

```diff
@@ -62,35 +61,43 @@ func NewInMemoryLRUCache[T any](opts ...InMemoryLRUCacheOpt[T]) *InMemoryLRUCach
 		opt(t)
 	}
 
-	t.ccache = ccache.New(ccache.Configure[T]().MaxSize(t.maxElements))
+	cache, _ := theine.NewBuilder[string, T](t.maxElements).Build()
+	t.cache = cache
 	return t
 }
 
 func (i InMemoryLRUCache[T]) Get(key string) T {
 	var zero T
-	item := i.ccache.Get(key)
-	if item == nil {
+	if i.cache == nil {
```

Comment:
Why are we checking for the existence of the cache in every method?

Reply:
The cache existence checks are needed because after Stop() is called, the cache becomes nil. The tests verify this behavior in [TestCachedCheckResolver_ResolveCheck_After_Stop_DoesNotPanic](https://github.com/Siddhant-K-code/openfga/blob/8c030dc5fff1610c86c47a1de2991559b36c94e0/internal/graph/cached_resolver_test.go#L646-L668), ensuring graceful handling when cache operations are attempted after stopping.

Reply:
The cache doesn't become `nil` externally. If anything you'd have to do `i.cache = nil` inside `Stop` but that would then require us to add an additional mutex.

That test you reference seems wrong since we wouldn't call `WithExistingCache` with `nil`. If that function accepts nil, it should actually panic.

There shouldn't be any `nil` checks for the methods. That test that we need to ensure behavior is, that the cache is never called after calling Stop. Which should only happen on cleanup/wrap up of the service.